pkg/controller/controller_instance.go

```diff
@@ -1071,6 +1128,30 @@ func (c *controller) updateServiceInstanceCondition(
 	return err
 }
 
+// prepareObservedGeneration sets the instance's observed generation
+// and clears the conditions, preparing it for any status updates that can occur
+// during the further processing.
+// It doesn't send the update request to server.
+func (c *controller) prepareObservedGeneration(toUpdate *v1beta1.ServiceInstance) {
+	toUpdate.Status.ObservedGeneration = toUpdate.Generation
+	reason := observedNewGenerationReason
+	message := observedNewGenerationMessage
+	setServiceInstanceCondition(
+		toUpdate,
+		v1beta1.ServiceInstanceConditionReady,
+		v1beta1.ConditionFalse,
+		reason,
+		message,
+	)
+	setServiceInstanceCondition(
+		toUpdate,
+		v1beta1.ServiceInstanceConditionFailed,
+		v1beta1.ConditionFalse,
```

Comment:
I'd argue that instead of setting the `Failed` condition to `false`, we should be removing the condition entirely. That seems the more Kube-way™ to me. For example, when transient errors occur on Deployments, they are tagged with the `ReplicaFailure` condition, but once they're resolved that condition is removed as opposed to unset. And I think it makes the user experience more straightforward during reattempts after spec edits, as the conditions a user would see wouldn't be different depending on whether they experienced a failure in the past, and they wouldn't have to walk through the logic of "oh, I have a Failed condition; wait, but it's false, so we're not actually failed".